### PR TITLE
Add debug logging and enhance block polling logic

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Added ability to set block limit for polling. Added additional logging for RPC method support checks and added safeguards for block data serialization.

### What changed?

- Added logging for RPC method support checks in `rpc.go`
- Introduced `POLL_UNTIL_BLOCK` environment variable to limit polling
- Modified block range calculation to prevent skipping the genesis block
- Updated `GetLatestPolledBlockNumber` to return `math.MaxUint64` when no block is found so next block polled will overflow to genesis
- Added a null check for `WithdrawalsHash` in block serialization

### How to test?

1. Set the `POLL_UNTIL_BLOCK` environment variable to test polling limits
2. Verify logging output for RPC method support checks
3. Ensure polling starts from genesis when no last polled block is found
4. Test block serialization with and without `WithdrawalsHash` data

### CODE REVIEW

**Comment by iuwqyir** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/19?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#19** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/19?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#18** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/18?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/17?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#16** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/16?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by AmineAfia — APPROVED** [review]

**Comment by iuwqyir** [comment]
### Merge activity

* **Sep 19, 5:02 AM EDT**: @iuwqyir started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/indexer/19).
